packages/ionic/src/commands/deploy/manifest.ts

```diff
@@ -31,10 +32,32 @@ export class DeployManifestCommand extends Command {
       throw new FatalException(`Cannot run ${input('ionic deploy manifest')} outside a project directory.`);
     }
 
-    const buildDir = path.resolve(this.project.directory, 'www'); // TODO: this is hard-coded
+    let buildDir: string;
+    if (this.project.getIntegration('capacitor') !== undefined) {
+      const conf = new CapacitorConfig(path.resolve(this.project.directory, CAPACITOR_CONFIG_FILE));
+      const webDir = conf.get('webDir');
+      if (webDir) {
+        buildDir = webDir;
+      } else {
+        throw new FatalException(`WebDir parameter has no value set in the Capacitor config file ${input(CAPACITOR_CONFIG_FILE)}`);
+      }
+    } else if (this.project.getIntegration('cordova') !== undefined) {
+      // for cordova it is hardcoded because www is mandatory
+      buildDir = path.resolve(this.project.directory, 'www');
```

Comment:
This should use `await this.project.getDistDir()`

Reply:
@dwieeb does `this.project.getDistDir()` return the correct dir for capacitor apps as well? If so I think this entire PR can be boiled down to that one line.

Reply:
it would appear not

Reply:
so maybe we can move all of this code into the `getDistDir` function itself?

Reply:
Yeah, probably. 

I would personally like to see more of what has been accomplished in `ionic deploy` put into lower-level code so it can be reused.

More examples:

- Projects will only be integrated with Cordova _or_ Capacitor (or neither), so that warrants a method in the `Project` class for getting the native integration (currently this exists in `deploy/core.ts`). 
- The Capacitor native project configuration logic (Info.plist and strings.xml manipulation) should probably be abstracted and put into `src/lib`. It's best to keep code in `src/commands` as lean as possible.

Reply:
Cool maybe we can make a separate refactor issue for future refactoring.